clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -118,24 +118,40 @@ protected static void adjustSeekCheckPointsBasedOnHeartbeats(
       }
       PubSubPosition heartbeatTimestampPosition =
```

Comment:
Maybe later: we can fetch offsets for all partitions in a single call to PubSub and process them together. We should also consider adding retries, in my opinion.